contracts/AcceleratingDistributor.sol

```diff
@@ -147,16 +145,24 @@ contract AcceleratingDistributor is Testable, ReentrancyGuard, Ownable, Multical
     }
 
     /**
-     * @notice Recover an ERC20 token either dropped on the contract or excess after the staking program ends.
-     * @dev Any wallet can call this function as it will only ever send tokens to the owner of the distributor.
+     * @notice Recover an ERC20 token either dropped on the contract or excess after the end of the staking program ends.
```

Comment:
Will we be able to recover staking tokens after the end of the program if staking tokens are explicitly disallowed from being withdrawn?

I think there's a tradeoff here:
1. Owner has "withdrawability", but the owner isn't _committing_ to paying out the tokens they contribute as they can always withdraw them. We could get this by allowing the staking token to be withdrawn but making this method `onlyOwner`.
2. Owner has less flexibility. Any tokens they contribute are forever committed to being paid out. However, users can be assured that the tokens will be paid out to stakers.

Which do we want?

Reply:
ok I have refactored this. we now have two separate methods:
1) `skim` that lets you extract any extra tokens on the contract that are NOT the reward token. this lets us recover tokens mistakenly dropped onto the contract, in the same way that you would in a uniswap pool. This is callable by anyone but always sends the tokens back to the `owner`.
2) `recoverReward` which lets the owner drain reward tokens from the contract at any point in time. The functionality in this is somewhat contentious as it lets the owner of the reward contract decide unilaterally to stop the reward program. it does not place the deposits funds at risk of being stolen or anything but it does change the credibility assumptions of the contract; before whatever was committed to the reward program was there until it ran out but now it can be drained if the owner changes their mind and wants to roll back their rewards.